scripts/ssh-hostkey.nse

```diff
@@ -316,24 +345,30 @@ local function portaction(host, port)
       bits=key.bits,
       key=key.key,
     }
-    if format:find( 'hex', 1, true ) or all_formats then
-      table.insert( output, ssh1.fingerprint_hex( key.fingerprint, key.algorithm, key.bits ) )
-    end
-    if format:find( 'bubble', 1, true ) or all_formats then
-      table.insert( output, ssh1.fingerprint_bubblebabble( openssl.sha1(key.fp_input), key.algorithm, key.bits ) )
+    -- Check for bad keys	
+    local bad_key=bin.pack("H",out.fingerprint)
+    if result[bad_key]~= out.fingerprint then
+    	if format:find( 'hex', 1, true ) or all_formats then
+      		table.insert( output, ssh1.fingerprint_hex( key.fingerprint, key.algorithm, key.bits ) )
+    	end
+    	if format:find( 'bubble', 1, true ) or all_formats then
+      		table.insert( output, ssh1.fingerprint_bubblebabble( openssl.sha1(key.fp_input), key.algorithm, key.bits ) )
+    	end
+    	if format:find( 'visual', 1, true ) or all_formats then
+      		table.insert( output, ssh1.fingerprint_visual( key.fingerprint, key.algorithm, key.bits ) )
+    	end
+    	if nmap.verbosity() > 1 or format:find( 'full', 1, true ) or all_formats then
+      		table.insert( output, key.full_key )
+    	end
+    	setmetatable(out, {
+        	__tostring = function(self)
+          	return table.concat(output, "\n")
+        	end
+      	})
+    	table.insert(output_tab, out)	
+    else
+    	stdnse.debug2("%s is a bad key.", out.fingerprintfile)
```

Comment:
"bad key" might be the wrong way to list this. Users might provide their own key lists for their own purposes. Using a format that includes room for a note could accommodate both use cases. Then we'd just output something like "known key found: <fingerprint>: <note>"